planner/core/preprocess.go

```diff
@@ -1696,13 +1696,7 @@ func (p *preprocessor) ensureInfoSchema() infoschema.InfoSchema {
 	if p.InfoSchema != nil {
 		return p.InfoSchema
 	}
-	// `Execute` under some conditions need to see the latest information schema.
-	if p.PreprocessExecuteISUpdate != nil {
-		if newInfoSchema := p.ExecuteInfoSchemaUpdate(p.Node, p.ctx); newInfoSchema != nil {
-			p.InfoSchema = newInfoSchema
-			return p.InfoSchema
-		}
-	}
+
```

Comment:
After this fix some codes will not be used any more, you can remove them too:

https://github.com/pingcap/tidb/blob/0f85650aa43cb60a77ce507e02203db45584e914/planner/core/preprocess.go#L79-L83
https://github.com/pingcap/tidb/blob/0f85650aa43cb60a77ce507e02203db45584e914/planner/optimize.go#L68-L88

Reply:
Done.